services/templates/service.go

```diff
@@ -117,20 +118,21 @@ func (s service) UpdateTemplate(c context.Context, template *entities.Template)
 }
 
 // DeleteTemplate deletes the given template
-func (s service) DeleteTemplate(c context.Context, template *entities.Template) error {
-	obj, err := s.s3.DeleteObject(&s3.DeleteObjectInput{
-		Bucket: aws.String(os.Getenv("TEMPLATE_BUCKET")),
-		Key:    aws.String(fmt.Sprintf("%d/%d", template.UserID, template.ID)),
-	})
+func (s *service) DeleteTemplate(c context.Context, templateID, userID int64) error {
+	_, err := s.db.GetTemplate(templateID, userID)
```

Comment:
No need to fetch the template here. We can simply perform the delete operation, since it won't be of any consequence if the record is not found.

Reply:
Agreed